OpenRA.Game/Traits/World/Shroud.cs

```diff
@@ -383,13 +453,25 @@ public bool IsVisible(Actor a)
 
 		public bool IsTargetable(Actor a)
 		{
+			IEnumerable<CPos> vis = GetVisOrigins(a);
+			bool concealed = vis.All(c => generatedDisruptionCount[c] > 0 || generatedShroudCount[c] > 0);
```

Comment:
If this is for targeting, then your "occupied cells" in the frozen actor need to instead store and deal with the targetable positions returned by the actor.  These are conceptually different to the occupied cells, even though they may (for the current set of traits) be derived from the same data.